### ISSUE TITLE
Unnumbered placeholder with multiple instances iterated over multiple times

### ISSUE THREAD

**Issue by randrej** [neoclide/coc.nvim#1707]
Thanks for enabling unnumbered placeholders!

Still, there's another issue with the current implementation: it iterates over every instance of an unnumbered placeholder. When vscode comes upon the first instance of a placeholder that repeats, it does a multiple-selection of all the instances and you may edit them all at once.

Since vim doesn't support multiple selection, I suggest only iterating over first instances of named placeholders. If you have time and will, you could make it so that the other instances are highlighted while editing the first one.

## Example
Test snippet:

```json
{
	"for": {
		"prefix": "for",
		"body": [
			"for (${size_t} ${i} = ${1:0}; ${i} < ${2:length}; ${i}++)",
			"{",
			"    $3",
			"}"
		],
		"description": "Code snippet for 'for' loop"
	}
}
```

### Desired behavior
When you expand this snippet in vscode and walk through the placeholders, it visits:

1. `0`
2. `length`
3. *the inside of code block*
4. `size_t`
5. `i` (first instance; if you're fancy, all instance highlighted)
6. *end of snippet*

### Observed behavior
When you expand this snippet in coc.nvim and walk through the placeholders, it visits:
1. `0`
2. `length`
3. *the inside of code block*
4. `size_t`
5. `i` first instance
6. `i` second instance
7. `i` third instance
8. *end of snippet*

**Comment by chemzqm** [comment]
Fixed by 761667b3

**Comment by randrej** [comment]
Thanks!
